R/table.R

```diff
@@ -3,21 +3,34 @@ setClass("Table", slots = list(name = "character"))
 
 #' Refer to a table nested in a hierarchy (e.g. within a schema)
 #'
+#' Objects of class `Table` have a single slot `name`, which is a named
+#' character vector.
+#'
 #' @param ... Components of the hierarchy, e.g. `schema`, `table`,
 #'   or `cluster`, `catalog`, `schema`, `table`.
 #'   For more on these concepts, see
 #'   \url{http://stackoverflow.com/questions/7022755/}
+#' @export
 Table <- function(...) {
-  new("Table", name = c(...))
+  components <- c(...)
+  if (is.null(names(components)) || any(names(components) == "")) {
```

Comment:
I am not sure we want to force naming, particularly for the most common case (schema & table).

Reply:
Which comes first -- schema or table? If we force naming, that confusion won't arise.

Reply:
Yes, but it makes it much more cumbersome to use the function. If it is cumbersome to use no one is going to use it. Having `Table("foo", "bar")` map to `'foo'.'bar'` seems like an easy solution, and if your backend uses different names you can instead use named arguments.

```r
if (length(components) == 2) && (is.null(names(components)) || all(names(components) == ""))) {
  names(components) <- c("schema", "table")
}
```

Reply:
Backends are free to implement their own wrappers with named arguments, and probably should too.

For instance, in BigQuery, "schema" means something entirely different. I'd rather keep things strict for the first implementation, we can always relax later but it's more difficult to impose restrictions on a function that has been exported.